sources/api/migration/migrations/v1.14.1/shibaken-aws-partition/src/main.rs

```diff
@@ -0,0 +1,26 @@
+use migration_helpers::common_migrations::{AddMetadataMigration, NoOpMigration, SettingMetadata};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a `setting-generator` for `settings.aws.partition` on AWS variants.
+/// This migration will do nothing on upgrade, but will remove the metadata if present on downgrade.
+fn run() -> Result<()> {
+    if cfg!(variant_platform = "aws") {
+        migrate(AddMetadataMigration(&[SettingMetadata {
+            setting: "settings.aws.partition",
+            metadata: &["setting-generator"],
+        }]))
+    } else {
+        migrate(NoOpMigration)
```

Comment:
This should be fine for non-AWS variants though, since there is no concept of a partition when it comes to IAM Roles Anywhere.

Reply:
Like the region behavior in `early-boot-config`, partition should fallback to `aws` if IMDS fails to pull one.